feat(tile): PF1:1

zeroedin opened the pull request:
### What I did

1. Added BaseTile class to inherit from
2. Added `selected` attribute to highlight the tile
3. Added `stacked` attribute which has two sizes `md` and `lg` to align the icon above the title

### Testing Instructions
Load up the deploy preview for this branch
Load up [PatternFly v4 Tabs](https://www.patternfly.org/v4/components/tile/)
Compare and contrast
Read over the [docs](https://deploy-preview-2187--patternfly-elements.netlify.app/components/tile/) and see if anything's missing

bennypowers commented:
@marionnegp this needs a design review. Do we need to block this on design alignment with upstream?

marionnegp reviewed (commented):
No need to hold off on `pfe-tile`. It's only `rh-tile` that's still being reviewed.

- [ ] Do we need the version where the icon isn't stacked (https://www.patternfly.org/v4/components/tile/#with-icon)?
- [ ] Border color should be #d2d2d2
- [x] Subtext color should be #151515
- [ ] Stacked icon should be 1.5rem

zeroedin commented:
@marionnegp On not having the stacked version, yes this was a regression.  `stacked` as an attribute when given should display an `md` if no `md | lg` size  is given, but the code was written accidently to always apply the `md` size which caused it always to be stacked which is incorrect.  Good catch.  Fixed in https://github.com/patternfly/patternfly-elements/pull/2187/commits/082e964246e07ed87e3fd11981853591e6f7d914

Colors and icon size also fixed in [06c7253](https://github.com/patternfly/patternfly-elements/pull/2187/commits/06c7253be18ee47aed45cebd47a64bee5c09f4ef)

bennypowers commented on .changeset/ten-keys-begin.md:
```diff
@@ -0,0 +1,10 @@
+"@patternfly/pfe-tile": major
+---
+
+Implements `<pfe-tile>` aligned with [PatternFly v4](https://patternfly.org/components/tile)
```
```suggestion
Implemented `<pfe-tile>` aligned with [PatternFly v4](https://patternfly.org/components/tile)
```

bennypowers commented on .changeset/ten-keys-begin.md:
```diff
@@ -0,0 +1,10 @@
+Implements `<pfe-tile>` aligned with [PatternFly v4](https://patternfly.org/components/tile)
+
+
+- Adds BaseTile and pfe-tile
```
```suggestion
- Added `BaseTile` and `<pfe-tile>`
```

bennypowers commented on .changeset/ten-keys-begin.md:
```diff
@@ -0,0 +1,10 @@
+
+
+- Adds BaseTile and pfe-tile
+- Added `selected` attribute to highlight the tile
```
these two lines should be snippets or in the docs instead of "added" - everything here is "added", right?

bennypowers commented on elements/pfe-tile/BaseTile.ts:
```diff
@@ -0,0 +1,20 @@
+import { LitElement, html } from 'lit';
+
+
+export abstract class BaseTile extends LitElement {
```
should have `@part` and `@slot` annotations

bennypowers commented on elements/pfe-tile/README.md:
```diff
@@ -0,0 +1,11 @@
+# Tile
+Add a description of the component here.
```
please elaborate

bennypowers commented on elements/pfe-tile/custom-elements-manifest.config.js:
```diff
@@ -0,0 +1,5 @@
+import { pfeCustomElementsManifestConfig } from '@patternfly/pfe-tools/custom-elements-manifest.js';
+
+export default pfeCustomElementsManifestConfig({
+globs: ['pfe-*.ts'],
```
also BaseTile

bennypowers commented on elements/pfe-tile/demo/demo.css:
```diff
@@ -0,0 +1,59 @@
+  max-width: 1008px;
+}
+
+.pf-l-flex {
```
best to avoid BEM classes IMO

bennypowers commented on elements/pfe-tile/docs/pfe-tile.md:
```diff
@@ -0,0 +1,109 @@
+{% renderOverview %}
+<div>
```
do we need the wrapper div?

bennypowers commented on elements/pfe-tile/docs/pfe-tile.md:
````diff
@@ -0,0 +1,109 @@
+</pfe-tile>
+```
+
+### Disabled
````
should go in `attributes`. probably stacked as well

bennypowers commented on elements/pfe-tile/package.json:
```diff
@@ -0,0 +1,50 @@
+    "test": "tests"
+  },
+  "exports": {
+    ".": "./pfe-tile.js",
```
BaseTile

bennypowers commented on elements/pfe-tile/pfe-tile.scss:
```diff
@@ -0,0 +1,123 @@
+:host {
+  position: relative;
+  display: inline-grid;
+  padding: var(--pf-c-tile--PaddingTop, var(--pf-global--spacer--lg, 1.5rem))
```
clean, clear, and under control :D

bennypowers commented on elements/pfe-tile/demo/pfe-tile.html:
```diff
@@ -0,0 +1,117 @@
+  <pfe-tile selected>
+    <svg slot="icon" fill="currentColor" height="1em" width="1em" viewBox="0 0 448 512" aria-hidden="true" role="img" style="vertical-align: -0.125em;"><path d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32 ...
+    Selected
+    <div slot="body">Subtext goes here</div>
```
should be <p>

bennypowers commented on elements/pfe-tile/demo/pfe-tile.html:
```diff
@@ -0,0 +1,117 @@
+  <pfe-tile>
+    <pfe-icon slot="icon" set="fas" icon="plus" size="md" loading="eager"></pfe-icon>
+    Default
+    <div slot="body">Subtext goes here</div>
```
```suggestion
    <p slot="body">Subtext goes here</p>
```

bennypowers reviewed (approved):
Lotsa Good sTuff, Mate
